src/unittests/f32_interval.spec.ts

```diff
@@ -300,134 +304,990 @@ g.test('contains_interval')
+    { input: [1, 2, 3], expected: false },
+    { input: [1, 2, 3, 4], expected: false },
+
+    // IntervalBounds
+    { input: [[1], [2]], expected: false },
+    { input: [[1], [2], [3]], expected: false },
+    { input: [[1], [2], [3], [4]], expected: false },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+      ],
+      expected: false,
+    },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+      ],
+      expected: false,
+    },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+        [4, 5],
+      ],
+      expected: false,
+    },
+
+    // F32Interval, valid dimensions
+    { input: [toF32Interval([1]), toF32Interval([2])], expected: true },
+    { input: [toF32Interval([1]), toF32Interval([2]), toF32Interval([3])], expected: true },
+    {
+      input: [toF32Interval([1]), toF32Interval([2]), toF32Interval([3]), toF32Interval([4])],
+      expected: true,
+    },
+
+    // F32Interval, invalid dimensions
+    { input: [toF32Interval([1])], expected: false },
+    {
+      input: [
+        toF32Interval([1]),
+        toF32Interval([2]),
+        toF32Interval([3]),
+        toF32Interval([4]),
+        toF32Interval([5]),
+      ],
+      expected: false,
+    },
+
+    // Mixed
+    { input: [1, [2]], expected: false },
+    { input: [1, [2], toF32Interval([3])], expected: false },
+    { input: [1, toF32Interval([2]), [3], 4], expected: false },
+    { input: [toF32Interval(1), 2], expected: false },
+    { input: [toF32Interval(1), [2]], expected: false },
+  ])
+  .fn(t => {
+    const input = t.params.input;
+    const expected = t.params.expected;
+
+    const got = isF32Vector(input);
+    t.expect(got === expected, `isF32Vector([${input}]) returned ${got}. Expected ${expected}`);
+  });
+
+interface ToF32VectorCase {
+  input: (number | IntervalBounds | F32Interval)[];
+  expected: (number | IntervalBounds)[];
+}
+
+g.test('toF32Vector')
+  .paramsSubcasesOnly<ToF32VectorCase>([
+    // numbers
+    { input: [1, 2], expected: [1, 2] },
+    { input: [1, 2, 3], expected: [1, 2, 3] },
+    { input: [1, 2, 3, 4], expected: [1, 2, 3, 4] },
+
+    // IntervalBounds
+    { input: [[1], [2]], expected: [1, 2] },
+    { input: [[1], [2], [3]], expected: [1, 2, 3] },
+    { input: [[1], [2], [3], [4]], expected: [1, 2, 3, 4] },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+      ],
+      expected: [
+        [1, 2],
+        [2, 3],
+      ],
+    },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+      ],
+      expected: [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+      ],
+    },
+    {
+      input: [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+        [4, 5],
+      ],
+      expected: [
+        [1, 2],
+        [2, 3],
+        [3, 4],
+        [4, 5],
+      ],
+    },
+
+    // F32Interval
+    { input: [toF32Interval([1]), toF32Interval([2])], expected: [1, 2] },
```

Comment:
Would be nice if some of these provided something other than a single `number` to `toF32Interval`